popmon/visualization/templates/card.html

```diff
@@ -1,16 +1,43 @@
-<div class="col-md-{% if 'full_width' in metric %}12{% else %}6{% endif %} mb-5">
+<div class="col-md-{% if 'full_width' in metric %}12{% else %}6{% endif %} mb-5" >
     <a name="{% if feature%}{{ feature.name }}-{%endif%}{{ metric.name }}"></a>
     <div class="card shadow-sm">
-        <div class="card-body" style="text-align: center">
+        <div class="card-body" id="{% if feature%}{{ feature.name }}-{%endif%}{{ metric.name }}-card" style="text-align: center">
             <h4 class="card-title">{{metric.name | fmt_metric}}</h4>
             {% if metric.description|length %}
                 <p class="card-text">{{metric.description}}</p>
             {% endif %}
         </div>
-        {% if 'table' in metric.plot %}
+        <img id="{% if feature%}{{ feature.name }}-{%endif%}{{ metric.name }}-loading" width="100" height="100" style="position: relative; display: inline-block; left: 50%; transform: translate(-50%);" src="https://c.tenor.com/8h6jKqp6ORgAAAAM/loading.gif" alt="Loading..." />
+        <div id= "{% if feature%}{{ feature.name }}-{%endif%}{{ metric.name }}"> </div>
+        {% if (('Overview' in section_title) or ('traffic_light' in metric.type)) %}
+            <script>
+                document.getElementById("{% if feature%}{{ feature.name }}-{%endif%}{{ metric.name }}-loading").style.display = "none";
+            </script>
             {{ metric.plot }}
         {% else %}
-        <img class="card-img-top" src="data:image/png;base64,{{metric.plot}}" alt="" />
-        {% endif %}
+            <script>
+                var observer = new IntersectionObserver(function(entries) {
+                    if(entries[0].isIntersecting === true){
+                        console.log("plotting graph");
+                        TESTER = document.getElementById("{% if feature%}{{ feature.name }}-{%endif%}{{ metric.name }}");
+                        var layout = feature{{ section_index }}{{ curr }}_layout["{{ metric.type }}"] ;
+                        {% if 'barplot' in metric.type %}
+                            delete layout["shapes"];
+                            delete layout["yaxis"]["range"];
+                        {% endif %}
+                        {% if metric.shapes | length %}
+                            var shapes = JSON.parse('{{ metric.shapes | tojson }}') ;
```

Comment:
JSON.parse should not be needed here, we could just provide the javascript object directly.